source/ShipModel.cpp

```diff
@@ -88,8 +88,34 @@ bool ShipModel::createAllTask(int data) {
 
 bool ShipModel::failAllTask() {
 	for (int i = 0; i < donuts.size(); i++) {
-		float angle = (float)(rand() % (int)(getSize()));
-		donuts.at(i)->setTeleportAngle(angle);
+		float newAngle = 0;
+		bool goodAngle = false;
+		while (!goodAngle) {
```

Comment:
Is it possible for the ship to get so full that `goodAngle` becomes permanently false? Should we hard code in a fail-safe against that?

Reply:
should be fine if we keep the MIN_DIFF small. I could add a loop counter and just accept whatever angle after x number of attempts

Reply:
That said, the same problem would exist in GLADOS, which is run more frequently :/

Reply:
I thought `GLaDOS` just skips the frame if it can't find a good angle on that frame?

Reply:
Hm okay